app/serializers/api/admin/order_serializer.rb

```diff
@@ -16,6 +16,12 @@ def distributor_name
     object.distributor.andand.name
   end
 
+  def display_outstanding_balance
+    return "" unless ["balance_due", "credit_owed"].include?(object.payment_state)
```

Comment:
There's some biz logic here that would be better in a service. if we do that, we could, at a later point, add to that service the logic in OrderUpdater#infer_payment_state).

It's not mandatory for this PR but ideally we would have this line like
    return "" unless PaymentStateService(object).outstanding_balance?

And then, in that new service, we would handle the complexity of outstanding balance versus payment_state.

I wrote this as a comment, not in scope for this PR, BUT then I went to the UK live database and checked these fields...
- I see 25 orders where payment_state is **credit_owed** and outstanding balance is 0
- I see 29 orders where payment_state is **balance_due** and outstanding balance is negative (payment total > total)
- I see 76 completed orders where payment_state is **failed** and there is a balance due (payment total < total)

I think we need to ignore payment_state in this code and just use object.outstanding_balance != 0

Reply:
Right so it seems like there are a whoooole bunch of edge cases I've clearly failed to think of.

Yeah this is much better, I've just had a wee hunt around the code base and it seems we do something similar here https://github.com/openfoodfoundation/openfoodnetwork/blob/b568d254467c4df1c891b917104599af3a4fdc97/app/views/spree/admin/payments/index.html.haml#L15 so there is definitely some scope to add this kind of logic to a PaymentStateService. But as you say it might be out of scope for this issue.

Reply:
@luisramos0  So I've just been testing this out locally with a few more order/payment combinations
![Screen Shot 2020-05-03 at 2 21 45 pm](https://user-images.githubusercontent.com/1100707/80898687-9038ec00-8d49-11ea-96bc-5e426bd5fe8a.png)

Seems to work as expected, I'm just wondering about the 'Failed' Payment state and whether it makes sense to have the outstanding balance shown there or not? What do you think?

Reply:
I like this table.
I think we should show the balance whenever it is not zero, regardless of the payment_state.

Reply:
I've updated the condition to not use payment state. Thanks for the input @luisramos0